Templates/templates/CSharp/Requests/IMethodCollectionPage.cs.tt

```diff
@@ -10,7 +10,7 @@ var interfaceConverterTypeName = @namespace.GetCoreLibraryType("InterfaceConvert
 var iCollectionPageTypeName = @namespace.GetCoreLibraryType("ICollectionPage");
 var iBaseClientTypeName = @namespace.GetCoreLibraryType("IBaseClient");
 
-var returnType = method.ReturnType.GetTypeString(@namespace);
+var returnType = method.ReturnType.GetTypeString(@namespace).DisambiguateRequestObject();
```

Comment:
The only language this can happen is Java and it doesn't use this disambiguation pattern as far as I can tell. @baywet can you confirm.

Reply:
it's not using a disambiguation as far as I can tell. But I don't believe it's required due to the way it's generating names (getting to the entity's/property's name and appending in cascade for all the different types)
Did you run into issues with java as well or only with CSharp?

Reply:
That's what I checked too.

We should create an end-to-end pipeline for Java as well so that I can quickly test scenarios similar to this. I will test it locally now to make sure that it builds fine.